bin/build.sh

```diff
@@ -1,13 +1,74 @@
 #!/usr/bin/env bash
 
+# This script is designed to build images and store package lists in the
+# following scenarios:
+# - Local builds with Docker Desktop and the `containerd` snapshotter. In this
+#   mode, all resulting images (multi-arch and single arch) will be loaded
+#   into the local container store. The `default` and `docker-container` drivers
+#   both work in this mode.
+# - For CI tests and package list generation with Docker and the `default`
+#   Docker driver. In this mode, resulting images will be loaded into the local
+#   container store, but only for amd64. Linux docker is not able to
+#   store/retreive multi-arch images locally. The `docker-container` driver will
+#   not work in this mode (it can't load images from the local store).
+# - Publishing images in CI with Docker and the `docker-container`
+#   driver. Pass in a REPO and PUBLISH_SUFFIX argument to publish images directly
+#   during the build. Since Docker is unable to store/reference multi-arch
+#   images locally, the publish process involves building/pushing an image to a
+#   disposable tag, then retagging it. The `default` Docker driver will not
+#   work in this mode (it can't build cross-architecture / multi-arch).
+
 set -euo pipefail
 
 cd "$(dirname "${BASH_SOURCE[0]}")/.."
 . bin/stack-helpers.sh
 
-[ $# -eq 1 ] || abort "usage: $(basename "${BASH_SOURCE[0]}") STACK_VERSION"
-
 STACK_VERSION=$1
+REPO=${2:-"heroku/heroku"}
+PUBLISH_SUFFIX=${3:-}
+
+[[ $STACK_VERSION == +([0-9]) ]] || abort "usage: $(basename "${BASH_SOURCE[0]}") STACK_VERSION [IMAGE_REPO] [PUBLISH_SUFFIX]"
```

Comment:
Should maybe be something like this instead?

```suggestion
[[ $STACK_VERSION =~ ^[0-9]+$ ]] || abort "usage: $(basename "${BASH_SOURCE[0]}") STACK_VERSION [IMAGE_REPO] [PUBLISH_SUFFIX]"
```

Reply:
Nice, thanks. Fixed in [8b9c5b2](https://github.com/heroku/base-images/pull/245/commits/8b9c5b21abed2b5e963fd67b55616643ee1a5c8e).